File: programme/validate-palindrome/validate-palindrome.c
// Write a programme to validate palindrome
// We assume that given input string is all lowered letters and also that the string has no spaces
// Because it is a char to char comparison it also works if we give a number as string input

#include <stdio.h>
#include <string.h>

// A function to check if a given string is palindrome
void isPalindrome(char input[])
{
  // 2 variables, in order to access first and last chars of input array
  int firstCharOfArray = 0;
  int lastCharOfArray = strlen(input) - 1;

  // Keep comparing characters while there are characters
  while (lastCharOfArray > firstCharOfArray)
  {
    if (input[firstCharOfArray++] != input[lastCharOfArray--])
    {
      printf("%s is not a palindrome\n\n", input);
      return;
    }
  }
  printf("%s is a palindrome\n\n", input);
}

// Script to check above code
int main()
{
  // Words Tests
  isPalindrome("coding");
  isPalindrome("level");
  isPalindrome("example");

  // Number Tests
  isPalindrome("121");
  isPalindrome("2022");

  // Mixed Tests
  isPalindrome("a1b2c2b1a");
  isPalindrome("123ab321");
}

File: programme/validate-palindrome/validate-palindrome.php
<?php
// Write a programme to validate palindrome

function Palindrome($MyString)
{
    $l = 0;
    $r = strlen($MyString) - 1;
    $flag = 0;

    while ($r > $l) {
        if ($MyString[$l] != $MyString[$r]) {
            $flag = 1;
            break;
        }
        $l++;
        $r--;
    }

    if ($flag == 0) {

        echo "True\n";
    } else {
        echo "False\n";
    }
}

$input = "radar";
echo "Input  : " . $input . "\n";
echo "Output : ";
Palindrome($input);
